operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java

```diff
@@ -67,6 +67,56 @@
 @SuppressWarnings("checkstyle:CyclomaticComplexity")
 public abstract class Ca {
 
+    /**
+     * A certificate entry in a ConfigMap. Each entry contains an entry name and data.
+     */
+    public enum SecretEntry {
+        /**
+         * A 64-bit encoded X509 Cretificate
+         */
+        CRT(".crt"),
+        /**
+         * Entity private key
+         */
+        KEY(".key"),
+        /**
+         * Entity certificate and key as a P12 keystore
+         */
+        P12_KEYSTORE(".p12"),
+        /**
+         * P12 keystore password
+         */
+        P12_KEYSTORE_PASSWORD(".password");
+
+        final String suffix;
+
+        SecretEntry(String suffix) {
+            this.suffix = suffix;
+        }
+
+        /**
+         * Retrieve a specific secret entry for pod from the given Secret.
+         * @param secret Kubernetes Secret containing desired entry
+         * @param podName Name of the pod which secret entry is looked for
+         * @param entry The SecretEntry type
+         * @return the data of the secret entry if found or null otherwise
+         */
+        public static String getDataForPod(Secret secret, String podName, SecretEntry entry) {
+            return secret.getData().get(getNameForPod(podName, entry));
+        }
+
+        /**
+         * Get the name of secret entry of given SecretEntry type for podName
+         * @param podName Name of the pod which secret entry is looked for
+         * @param entry The SecretEntry type
+         * @return the name of the secret entry
+         */
+        public static String getNameForPod(String podName, SecretEntry entry) {
+            return podName + entry.suffix;
+        }
```

Comment:
It seems a bit weird to have these in the Enum. Would it make more sense to have them in the Ca class?

Reply:
The idea here is to encapsulate all the logic around the name composition in a single type. I'm fine with moving it to the Ca as well given such Secret related naming logic exists (e.g. annotation names etc, which IMHO are a good candidate for being converted into an ENUM as well).

Lemme know your preference.

Reply:
For me, Enum should cover a finite set of options. You clearly have here (for various reasons - some valid, some probably just our stupidity) an infinite set of key names. So at least from my personal point of view -> keep the Enum where it applies and move the other methods away. Or - change it to a regular class with some constants and some methods to generate other names dynamically. IMHO, the main advantage of the Enum would be the type enforcement. But you do not have it anyway if you somewhere use the Enum and somewhere just a generated String.

Feel free to check with Tom Bentley if you want - he might have a better idea of how common is to mix unrelated methods into the Enum like this.

Reply:
Another problem of moving those methods outside is that the encapsulation of the suffix would break, since this code relies on the knowledge that there's a "suffix" , which will need to have a getter and IMHO it would look ugly.

However from the perspective of this Enum usage, this code makes more sense to me:

```java
String store = SecretEntry.getDataForPod(secret, podName, SecretEntry.P12_KEYSTORE);
String password = SecretEntry.getDataForPod(secret, podName, SecretEntry.P12_KEYSTORE_PASSWORD);
```

Compared to e.g.

```java
String store = Ca.getSecretDataForPod(secret, podName, SecretEntry.P12_KEYSTORE);
String password = Ca.getSecretDataForPod(secret, podName, SecretEntry.P12_KEYSTORE_PASSWORD);
```

Again, it's just an improvement which I can revert and hard code the names as they were.

Reply:
The two code snippets look almost identical to me. I do not see anything broken about it.

Reply:
I thought maybe it's easier if we talk about a specific code so I've pushed a commit with the refactoring to a separate branch https://github.com/devguyio/strimzi-kafka-operator/commit/24db7e751ae447cba0732da6f9b20de7c5058303

Is that the change you feel we should go with?

Reply:
That code looks mostly good to me. I would move the Enum up to p or down to the bottom of the class. A nd the methods with other methods. But that is just the placement in the file. The code it self would work for me.

Reply:
ack, I just used the IDE to refactor and get a change out. Will change it properly and push it. Thanks!

Reply:
Done